### DISCUSSION TITLE
Need CLLE prompting

### DISCUSSION THREAD

**Discussion by NextPage66** [codefori/vscode-ibmi#1972] [Q&A] [answered]
I have installed 2 extensions, CLLE language tools (IBM) and CL/CLLE (SpineNetSystems).  I need a way to prompt the CL commands and a way to look for the correct spelling of some commands.
Neither of these seem to help in this area.
What am I doing wrong?
Is there another extension that will do what I need?
 

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.8.0|
|Visual Studio Code version|1.87.2|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
CL (vscode-clle): 1.1.4
COBOL (cobol): 10.3.10
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.4.0
Db2 for IBM i (vscode-db2i): 0.8.2
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.16.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
IBM i Renderer (vscode-displayfile): 0.1.2
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Pylance (vscode-pylance): 2024.3.1
Python (python): 2024.2.1
Python Test Explorer for Visual Studio Code (vscode-python-test-adapter): 0.8.2
RPGLE (vscode-rpgle): 0.26.3
Remote - SSH (remote-ssh): 0.109.0
Remote Explorer (remote-explorer): 0.4.3
SQLTools (sqltools): 0.28.1
TODO Highlight (vscode-todo-highlight): 1.0.5
Test Adapter Converter (test-adapter-converter): 0.1.9
Test Explorer UI (vscode-test-explorer): 2.21.1
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|8|
|CCSID|37|
|Default CCSID|37|
|SQL|Enabled
|Source dates|Enabled

### Enabled features

|/usr/bin|/QOpenSys/pkgs/bin||/QSYS.LIB|
|-|-|-|-|
|attr|bash|ILEDITOR.GETNEWLIBL|QZDFMDB2.PGM|
|iconv|chsh|||
|setccsid|ls|||
|tar|md5sum|||
||sort|||
||stat|||
</details>

<details>
<summary>Shell env</summary>

```bash

_=/QOpenSys/pkgs/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF3012:  File ZIP in library *LIBL not found.\nCPC2206:  Ownership of object QZSHSYSTEM in QTEMP type *USRSPC changed.",
    "cwd": ""
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF3012:  File ZAC in library *LIBL not found.\nCPC2206:  Ownership of object QZSHSYSTEM in QTEMP type *USRSPC changed.",
    "cwd": ""
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF3012:  File ZKY in library *LIBL not found.\nCPC2206:  Ownership of object QZSHSYSTEM in QTEMP type *USRSPC changed.",
    "cwd": ""
  }
]
```
</details>

**Comment by worksofliam** [comment, marked as answer]
Ensure you have CL language tools version 1.1.5 installed. This version just came out.

Then, you can use the content assist in your CL code.

<img width="515" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/3708366/0034d615-ac3a-4d4c-a306-dfb1c37fd4b9">

VS Code does not support fully fledged CL prompting (yet.)

**Reply by heymchri** [reply]
@worksofliam sorry for this noob question - how do you get the content assist to show all the parameters like you're showing in the screenshot?

**Reply by worksofliam** [reply]
@heymchri control+space is the answer!

**Reply by NextPage66** [reply]
hyemchri, I had to go back and get the CL language tools 1.1.5 and then the control+space worked.  
Many Thanks to worksofliam.
